client/src/services/data-interface/data-interface.ts

```diff
@@ -0,0 +1,194 @@
+export interface BusinessLicense {
+  entity_number: string;
+  business_name: string;
+  dba_name: string | null;
+  address: string;
+  zipcode: BostonZipCode;
+  license_number: string;
+  status: string | null;
+  alcohol_type: string;
+  file_name: string;
+}
+
+type BostonZipCode =
+  | "02021"
+  | "02026"
+  | "02108"
+  | "02109"
+  | "02110"
+  | "02111"
+  | "02113"
+  | "02114"
+  | "02115"
+  | "02116"
+  | "02118"
+  | "02119"
+  | "02120"
+  | "02121"
+  | "02122"
+  | "02124"
+  | "02125"
+  | "02126"
+  | "02127"
+  | "02128"
+  | "02129"
+  | "02130"
+  | "02131"
+  | "02132"
+  | "02134"
+  | "02135"
+  | "02136"
+  | "02151"
+  | "02152"
+  | "02163"
+  | "02186"
+  | "02199"
+  | "02203"
+  | "02210"
+  | "02215"
+  | "02459"
+  | "02467";
+
+const validBostonZipCodes: Set<string> = new Set([
+  "02021",
+  "02026",
+  "02108",
+  "02109",
+  "02110",
+  "02111",
+  "02113",
+  "02114",
+  "02115",
+  "02116",
+  "02118",
+  "02119",
+  "02120",
+  "02121",
+  "02122",
+  "02124",
+  "02125",
+  "02126",
+  "02127",
+  "02128",
+  "02129",
+  "02130",
+  "02131",
+  "02132",
+  "02134",
+  "02135",
+  "02136",
+  "02151",
+  "02152",
+  "02163",
+  "02186",
+  "02199",
+  "02203",
+  "02210",
+  "02215",
+  "02459",
+  "02467",
+]);
+
+type ValidationResult =
+  | { valid: true; data: BusinessLicense }
+  | { valid: false; errors: Record<string, string> };
+
+function isBostonZipCode(zipcode: unknown): zipcode is BostonZipCode {
+  return typeof zipcode === "string" && validBostonZipCodes.has(zipcode);
+}
+
+function validateBusinessLicense(license: unknown): ValidationResult {
+  const errors: Record<string, string> = {};
+
+  if (typeof license !== "object" || license === null) {
+    return { valid: false, errors: { root: "Not an object or is null" } };
+  }
+
+  const obj = license as Record<string, unknown>;
+
+  if (typeof obj.entity_number !== "string") {
+    errors.entity_number = "Must be a string";
+  }
+
+  if (typeof obj.business_name !== "string") {
+    errors.business_name = "Must be a string";
+  }
+
+  if (obj.dba_name !== null && typeof obj.dba_name !== "string") {
+    errors.dba_name = "Must be a string or null";
+  }
+
+  if (typeof obj.address !== "string") {
+    errors.address = "Must be a string";
+  }
+
+  if (!isBostonZipCode(obj.zipcode)) {
+    errors.zipcode = "Must be a valid Boston zip code";
+  }
+
+  if (typeof obj.license_number !== "string") {
+    errors.license_number = "Must be a string";
+  }
+
+  if (obj.status !== null && typeof obj.status !== "string") {
+    errors.status = "Must be a string or null";
+  }
+
+  if (typeof obj.alcohol_type !== "string") {
+    errors.alcohol_type = "Must be a string";
+  }
+
+  if (typeof obj.file_name !== "string") {
+    errors.file_name = "Must be a string";
+  }
+
+  if (Object.keys(errors).length > 0) {
+    return { valid: false, errors };
+  }
+
+  return { valid: true, data: obj as unknown as BusinessLicense };
```

Comment:
🌴 Another non blocking comment: We lose some type safety here. If the structure of a license changes, additional fields could slip by unchecked. Considering the volunteer nature of the project, it would be good idea to alert developers that adding fields to the data would have effects here.

```suggestion
  const validatedBusinessLicense: BusinessLicense = {
    entity_number: obj.entity_number,
    business_name: obj.business_name,
    dba_name: obj.dba_name,
    address: obj.address,
    zipcode: obj.zipcode,
    license_number: obj.license_number,
    status: obj.status,
    alcohol_type: obj.alcohol_type,
    file_name: obj.file_name,
  }
  
  return { valid: true, data: validatedBusinessLicense }
```